Source/VMware.vSphereDSC/DSCResources/DatastoreClusterDatastore.ps1

```diff
@@ -0,0 +1,280 @@
+    Empty location means that the Datacenter is in the Root Folder of the Inventory.
+    The Folder names in the location are separated by '/'.
+    Example Datacenter location: 'MyDatacentersFolderOne/MyDatacentersFolderTwo'.
+    #>
+    [DscProperty(Key)]
+    [string] $DatacenterLocation
+
+    <#
+    .DESCRIPTION
+
+    Specifies the name of the Datastore Cluster located in the Datacenter specified in 'DatacenterName' key property.
+    #>
+    [DscProperty(Key)]
+    [string] $DatastoreClusterName
+
+    <#
+    .DESCRIPTION
+
+    Specifies the location of the Datastore Cluster with name specified in 'DatastoreClusterName' key property in the Datacenter
+    specified in 'DatacenterName' key property. Location consists of 0 or more Folders.
+    Empty location means that the Datastore Cluster is located in the Datastore Folder of the Datacenter.
+    The Root Folders of the Datacenter are not part of the location. Folder names in the location are separated by '/'.
+    Example location for a Datastore Cluster: 'MyDatastoreClusterFolderOne/MyDatastoreClusterFolderTwo'.
+    #>
+    [DscProperty(Key)]
+    [string] $DatastoreClusterLocation
+
+    <#
+    .DESCRIPTION
+
+    Specifies the names of the Datastores that should be located in the specified Datastore Cluster.
+    #>
+    [DscProperty(Mandatory)]
+    [string[]] $DatastoreNames
+
+    <#
+    .DESCRIPTION
+
+    Specifies the instance of the 'InventoryUtil' class that is used
+    for Inventory operations.
+    #>
+    hidden [InventoryUtil] $InventoryUtil
+
+    hidden [string] $AddDatastoresToDatastoreClusterMessage = "Adding Datastores {0} to Datastore Cluster {1}."
+
+    hidden [string] $CouldNotFindDatastoreMessage = "Could not find Datastore {0} in Datacenter {1}."
+    hidden [string] $CouldNotAddDatastoresToDatastoreClusterMessage = "Could not add Datastores {0} to Datastore Cluster {1}. For more information: {2}"
+
+    [void] Set() {
+        try {
+            Write-VerboseLog -Message $this.SetMethodStartMessage -Arguments @($this.DscResourceName)
+            $this.ConnectVIServer()
+
+            $this.InitInventoryUtil()
+            $datacenter = $this.InventoryUtil.GetDatacenter($this.DatacenterName, $this.DatacenterLocation)
+            $datacenterDatastoreFolderName = [FolderType]::Datastore.ToString() + 'Folder'
+            $datastoreCluster = $this.InventoryUtil.GetDatastoreCluster(
+                $this.DatastoreClusterName,
+                $this.InventoryUtil.GetInventoryItemParent(
+                    $this.DatastoreClusterLocation,
+                    $datacenter,
+                    $datacenterDatastoreFolderName
+                )
+            )
+
+            $datastores = $this.GetDatastores($datacenter)
+            $datastoresToAddToDatastoreCluster = $this.GetDatastoresToAddToDatastoreCluster($datastoreCluster, $datastores)
+
+            $this.AddDatastoresToDatastoreCluster($datastoreCluster, $datastoresToAddToDatastoreCluster)
```

Comment:
Why only Add is applied in this resource?
I my intent is to configure the datastore in a datastore cluster with this resource and I specify the list of desired datastores I'd expect to have the exact those after Set configuration. It seems this resource is just for the purpose of adding datastore.

Reply:
Renamed the DSC Resource to **DatastoreClusterAddDatastore** to better understand the covered use case.